stdlib/3.4/pathlib.pyi

```diff
@@ -1,164 +1,74 @@
-    def make_uri(self, path): ...
-
-class _Accessor: ...
-
-class _NormalAccessor(_Accessor):
-    stat = ... # type: Any
-    lstat = ... # type: Any
-    open = ... # type: Any
-    listdir = ... # type: Any
-    chmod = ... # type: Any
-    lchmod = ... # type: Any
-    mkdir = ... # type: Any
-    unlink = ... # type: Any
-    rmdir = ... # type: Any
-    rename = ... # type: Any
-    replace = ... # type: Any
-    def symlink(a, b, target_is_directory): ...
-    utime = ... # type: Any
-    def readlink(self, path): ...
-
-class _Selector:
-    child_parts = ... # type: Any
-    successor = ... # type: Any
-    def __init__(self, child_parts) -> None: ...
-    def select_from(self, parent_path): ...
-
-class _TerminatingSelector: ...
-
-class _PreciseSelector(_Selector):
-    name = ... # type: Any
-    def __init__(self, name, child_parts) -> None: ...
-
-class _WildcardSelector(_Selector):
-    pat = ... # type: Any
-    def __init__(self, pat, child_parts) -> None: ...
-
-class _RecursiveWildcardSelector(_Selector):
-    def __init__(self, pat, child_parts) -> None: ...
-
-class _PathParents(Sequence):
-    def __init__(self, path) -> None: ...
-    def __len__(self): ...
-    def __getitem__(self, idx): ...
+from typing import Any, Generator, IO, Optional, Sequence, Tuple, Union
+import os
 
 class PurePath:
-    def __init__(self, *args) -> None: ...
-    def __reduce__(self): ...
-    def as_posix(self): ...
-    def __bytes__(self): ...
-    def as_uri(self): ...
-    def __eq__(self, other): ...
-    def __ne__(self, other): ...
-    def __hash__(self): ...
-    def __lt__(self, other): ...
-    def __le__(self, other): ...
-    def __gt__(self, other): ...
-    def __ge__(self, other): ...
-    drive = ... # type: Any
-    root = ... # type: Any
-    @property
-    def anchor(self): ...
-    @property
-    def name(self): ...
-    @property
-    def suffix(self): ...
-    @property
-    def suffixes(self): ...
-    @property
-    def stem(self): ...
-    def with_name(self, name): ...
-    def with_suffix(self, suffix): ...
-    def relative_to(self, *other): ...
-    @property
-    def parts(self): ...
-    def joinpath(self, *args): ...
-    def __truediv__(self, key): ...
-    def __rtruediv__(self, key): ...
-    @property
-    def parent(self): ...
-    @property
-    def parents(self): ...
-    def is_absolute(self): ...
-    def is_reserved(self): ...
-    def match(self, path_pattern): ...
+    parts = ...  # type: Tuple[str, ...]
+    drive = ... # type: str
+    root = ... # type: str
+    anchor = ...  # type: str
+    parents = ...  # type: Sequence[PurePath]
+    parent = ...  # type: PurePath
+    name = ...  # type: str
+    suffix = ...  # type: str
+    suffixes = ...  # type: List[str]
+    stem = ...  # type: str
+    def __init__(self, *pathsegments: str) -> None: ...
+    def __hash__(self) -> int: ...
+    def __eq__(self, other: Any) -> bool: ...
+    def __ne__(self, other: Any) -> bool: ...
```

Comment:
Also, the other arg to `__eq__` and `__ne__` should have type `object`, not
`Any`.

On Thu, Jul 7, 2016 at 1:24 PM, Matthias Kramm <notifications@github.com>
wrote:

> In stdlib/3.4/pathlib.pyi
> <https://github.com/python/typeshed/pull/338#discussion_r69979629>:
>
> > -    def is_reserved(self): ...
> > -    def match(self, path_pattern): ...
> > +    parts = ...  # type: Tuple[str, ...]
> > +    drive = ... # type: str
> > +    root = ... # type: str
> > +    anchor = ...  # type: str
> > +    parents = ...  # type: Sequence[PurePath]
> > +    parent = ...  # type: PurePath
> > +    name = ...  # type: str
> > +    suffix = ...  # type: str
> > +    suffixes = ...  # type: List[str]
> > +    stem = ...  # type: str
> > +    def __init__(self, *pathsegments: str) -> None: ...
> > +    def __hash__(self) -> int: ...
> > +    def __eq__(self, other: Any) -> bool: ...
> > +    def __ne__(self, other: Any) -> bool: ...
>
> Are these necessary? Won't they be inherited from object?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/python/typeshed/pull/338/files/6dbb683f00b0652615e1f3b027453811be350a71#r69979629>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ACwrMsSdSH-RWWuFnkQuBZ-_gl9xwhcmks5qTWBvgaJpZM4JFIbK>
> .
>



-- 
--Guido van Rossum (python.org/~guido)